lib/resolve-imports-exports.js

```diff
@@ -0,0 +1,208 @@
+                + 'the "exports" resolution of ' + packagePath + path.sep + 'package.json imported from ' + parent + '.');
+            err.code = 'ERR_INVALID_MODULE_SPECIFIER';
+            throw err;
+        }
+
+        return resolved;
+    }
+
+    if (Array.isArray(target)) {
+        if (target.length === 0) {
+            err = new Error(key === '.'
+                ? 'No "exports" main resolved in ' + packagePath + path.sep + 'package.json.'
+                : 'Package subpath ' + key + ' is not defined by "exports" in ' + packagePath + path.sep + 'package.json imported from ' + parent + '.');
+            err.code = 'ERR_PACKAGE_PATH_NOT_EXPORTED';
+            throw err;
+        }
+
+        var lastError;
+        for (var i = 0; i < target.length; i++) {
+            try {
+                return resolvePackageTarget(packagePath, parent, key, target[i], subpath, internal, conditions);
+            } catch (e) {
+                if (e && (e.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED' || e.code === 'ERR_INVALID_PACKAGE_TARGET')) {
+                    lastError = e;
+                } else {
+                    throw e;
+                }
+            }
+        }
+        throw lastError;
+    }
+
+    if (target === null) {
+        err = new Error(key === '.'
+            ? 'No "exports" main resolved in ' + packagePath + path.sep + 'package.json.'
+            : 'Package subpath ' + key + ' is not defined by "exports" in ' + packagePath + path.sep + 'package.json imported from ' + parent + '.');
+        err.code = 'ERR_PACKAGE_PATH_NOT_EXPORTED';
+        throw err;
+    }
+
+    if (typeof target !== 'object') {
+        err = new Error('Invalid "exports" target ' + JSON.stringify(target)
+            + ' defined for ' + key + ' in ' + packagePath + path.sep + 'package.json imported from ' + parent + '.');
+        err.code = 'ERR_INVALID_PACKAGE_TARGET';
+        throw err;
+    }
+
+    var exportedConditions = validateConditions(Object.keys(target));
+
+    for (i = 0; i < exportedConditions.length; i++) {
+        var exportedCondition = exportedConditions[i];
+        if (exportedCondition === 'default' || conditions.indexOf(exportedCondition) !== -1) {
+            try {
+                return resolvePackageTarget(
+                    packagePath,
+                    parent,
+                    key,
+                    target[exportedCondition],
+                    subpath,
+                    internal,
+                    conditions
+                );
+            } catch (e) {
+                if (!e || e.code !== 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+                    throw e;
+                }
+            }
+        }
+    }
+
+    err = new Error(key === '.'
+        ? 'No "exports" main resolved in ' + packagePath + path.sep + 'package.json.'
+        : 'Package subpath ' + key + ' is not defined by "exports" in ' + packagePath + path.sep + 'package.json imported from ' + parent + '.');
```

Comment:
```suggestion
        ? 'No "exports" main resolved in ' + path.join(packagePath, 'package.json') + '.'
        : 'Package subpath ' + key + ' is not defined by "exports" in ' + path.join(packagePath, 'package.json') + ' imported from ' + parent + '.');
```
alternatively for all these, we could pre-join the packageJSONpath and use that in all these places